model-engine/model_engine_server/infra/gateways/resources/k8s_endpoint_resource_delegate.py

```diff
@@ -261,16 +299,59 @@ def add_datadog_env_to_main_container(deployment_template: Dict[str, Any]) -> No
     user_container["env"] = user_container_envs
 
 
+def add_lws_default_env_vars_to_container(container: Dict[str, Any]) -> None:
+    container_envs = []
+    container_envs.extend(
+        [
+            {"name": "K8S_OWN_POD_NAME", "valueFrom": {"fieldRef": {"fieldPath": "metadata.name"}}},
+            {
+                "name": "K8S_OWN_NAMESPACE",
+                "valueFrom": {"fieldRef": {"fieldPath": "metadata.namespace"}},
+            },
+            {
+                "name": "K8S_LWS_NAME",
+                "valueFrom": {
+                    "fieldRef": {"fieldPath": "metadata.labels['leaderworkerset.sigs.k8s.io/name']"}
+                },
+            },
+            {  # This may not do anything actually, prefer to use LWS_LEADER_ADDRESS instead
```

Comment:
is this comment still relevant?

Reply:
yeah it is, we should be using LWS_LEADER_ADDRESS in the vllm bundle

Reply:
should we remove the env var then?